File: ruby/bob/bob.rb
class Bob
  def self.hey(talk)
    talk = talk.strip
    silence = talk.empty?
    yelling = talk.upcase == talk && talk.match?(/[A-Z]/)
    asking = talk.end_with?('?')

    if silence then
      return "Fine. Be that way!"
    end

    if yelling and asking then
      return "Calm down, I know what I'm doing!"
    end

    if asking then
      return "Sure."
    end

    if yelling then
      return "Whoa, chill out!"
    end

    return "Whatever."
  end
end

File: fsharp/bob/Bob.fs
module Bob

open System.Text.RegularExpressions

let response (input: string): string =
    let input = input.Trim()
    let silence = input |> System.String.IsNullOrEmpty
    let yelling = input.ToUpper() = input && Regex(@"[A-Z]").IsMatch(input)
    let asking = Regex(@"\?$").IsMatch(input)
    match (silence, yelling, asking) with
    | (true, _, _) -> "Fine. Be that way!"
    | (_, true, true) -> "Calm down, I know what I'm doing!"
    | (_, _, true) -> "Sure."
    | (_, true, _) -> "Whoa, chill out!"
    | _ -> "Whatever."

    